sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/ParametersCorrectOrder.cs

```diff
@@ -47,90 +48,115 @@ protected sealed override void Initialize(SonarAnalysisContext context)
             context.RegisterSyntaxNodeActionInNonGenerated(
                 c =>
                 {
-                    var methodCall = (InvocationExpressionSyntax) c.Node;
-                    var methodParameterLookup = new MethodParameterLookup(methodCall, c.SemanticModel);
-                    var argumentParameterMappings = methodParameterLookup.GetAllArgumentParameterMappings()
-                        .ToDictionary(pair => pair.Argument, pair => pair.Parameter);
+                    var methodCall = (InvocationExpressionSyntax)c.Node;
 
-                    var methodSymbol = methodParameterLookup.MethodSymbol;
-                    if (methodSymbol == null)
-                    {
-                        return;
-                    }
+                    var memberAccess = methodCall.Expression as MemberAccessExpressionSyntax;
+                    Func<Location> getLocation = () =>
+                        memberAccess == null
+                        ? methodCall.Expression.GetLocation()
+                        : memberAccess.Name.GetLocation();
 
-                    var parameterNames = argumentParameterMappings.Values
-                        .Select(symbol => symbol.Name)
-                        .Distinct()
-                        .ToList();
+                    AnalyzeArguments(c, methodCall.ArgumentList, getLocation);
+                }, SyntaxKind.InvocationExpression);
 
-                    var identifierArguments = GetIdentifierArguments(methodCall);
-                    var identifierNames = identifierArguments
-                        .Select(p => p.IdentifierName)
-                        .ToList();
+            context.RegisterSyntaxNodeActionInNonGenerated(
+                c =>
+                {
+                    var objectCreationCall = (ObjectCreationExpressionSyntax)c.Node;
 
-                    if (!parameterNames.Intersect(identifierNames).Any())
-                    {
-                        return;
-                    }
+                    var qualifiedAccess = objectCreationCall.Type as QualifiedNameSyntax;
+                    Func<Location> getLocation = () =>
+                        qualifiedAccess == null
+                        ? objectCreationCall.Type.GetLocation()
+                        : qualifiedAccess.Right.GetLocation();

```

Comment:
I am not sure if these two are prone to errors due invalid syntax. Perhaps it is worth checking if they work with something like `new ()` or `new System. ()`...